ci/kokoro/docker/build.sh

```diff
@@ -226,16 +220,29 @@ elif [[ "${BUILD_NAME}" = "cxx17" ]]; then
 elif [[ "${BUILD_NAME}" = "quickstart-bazel" ]]; then
   export DISTRO=ubuntu
   export DISTRO_VERSION=18.04
+  RUN_INTEGRATION_TESTS="no" # quickstart does not support integration tests.
   in_docker_script="ci/kokoro/docker/build-in-docker-quickstart-bazel.sh"
 elif [[ "${BUILD_NAME}" = "quickstart-cmake" ]]; then
   export DISTRO=ubuntu
   export DISTRO_VERSION=18.04
+  RUN_INTEGRATION_TESTS="no" # quickstart does not support integration tests.
   in_docker_script="ci/kokoro/docker/build-in-docker-quickstart-cmake.sh"
 else
   echo "Unknown BUILD_NAME (${BUILD_NAME}). Fix the Kokoro .cfg file."
   exit 1
 fi
 
+# If RUN_INTEGRATION_TESTS wasn't set in the environment or by one of the
+# builds above, default it to "yes" for kokoro builds and "auto" otherwise.
+if [[ -z "${RUN_INTEGRATION_TESTS:-}" ]]; then
+  if [[ -n "${KOKORO_JOB_NAME:-}" ]]; then
```

Comment:
Rather than consulting `KOKORO_JOB_NAME` as a proxy for whether this is running as a CI build, you might use `RUNNING_CI`